src/d3a/models/strategy/storage.py

```diff
@@ -513,8 +513,7 @@ def buy_energy(self, market, offer=None):
             return
         if self.state.has_battery_reached_max_power(-FLOATING_POINT_TOLERANCE, market.time_slot):
             return
-        max_affordable_offer_rate = min(self.bid_update.get_updated_rate(market.time_slot),
-                                        self.bid_update.final_rate[market.time_slot])
+        max_affordable_offer_rate = self.bid_update.get_updated_rate(market.time_slot)
```

Comment:
Does not hurt to change that, however I am curious why this was needed in the first place. Essentially, what this did was a sanity check in order to always make sure that the bid rate will never surpass the final rate (in order to be able to cover potential problems with the rate calculation like e.g. border conditions due to floating point precision) , therefore I think it would be better if you left it at peace unless there is a specific reason to change it.

Reply:
sorry to jump in, Spyros AFAI, the sanity check is already part of [bid_update.get_updated_rate](https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/strategy/update_frequency.py#L135) implementation

Reply:
Then it is all good here, leave it as it is